File: Data Structures & Algorithms/pow-x-n/submission-0.cpp
class Solution {
public:
    double myPow(double x, int n) {

        if(n < 0){
            x = (double) 1/x;
            n = abs(n);
        }
        
        if(n == 0){
            return 1.00000;
        }

        if(n == 1){
            return x;
        }

        double ans = 0.0000;
        if(n%2){
            ans += myPow(x, n-1) * x;
        } else {
            double temp = myPow(x, n >> 1);
            ans += temp * temp;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/palindrome-partitioning/submission-0.java
class Solution {
    List<List<String>> ans = new ArrayList<>();
    private boolean isPalindrome(String str) {
        int l = 0, r = str.length() - 1;
        while (l <= r) {
            if (str.charAt(l) != str.charAt(r)) {
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
    private void helper(String str, List<String> l, int idx) {
        if (idx >= str.length()) {
            ans.add(new ArrayList<>(l));
            return;
        }
        String temp;
        for (int i = idx; i < str.length(); i++) {
            temp = str.substring(idx, i + 1);

            if (isPalindrome(temp)) {
                l.add(temp);
                helper(str, l, i + 1);
                l.remove(l.size() - 1);
            }
        }
    }
    public List<List<String>> partition(String s) {
        helper(s, new ArrayList<>(), 0);
        return ans;
    }
}
